hadoop-ozone/client/src/test/java/org/apache/hadoop/ozone/client/io/TestKeyOutputStream.java

```diff
@@ -0,0 +1,154 @@
+
+
+import org.apache.hadoop.hdds.client.BlockID;
+import org.apache.hadoop.hdds.client.RatisReplicationConfig;
+import org.apache.hadoop.hdds.client.ReplicationConfig;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.OzoneClientConfig;
+import org.apache.hadoop.hdds.scm.XceiverClientFactory;
+import org.apache.hadoop.hdds.scm.container.common.helpers.ExcludeList;
+import org.apache.hadoop.hdds.scm.pipeline.Pipeline;
+import org.apache.hadoop.hdds.scm.pipeline.PipelineID;
+import org.apache.hadoop.ozone.om.helpers.OpenKeySession;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.protocolPB.OzoneManagerClientProtocol;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.ReplicationFactor.THREE;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test KeyOutputStream with RATIS keys.
+ */
+public class TestKeyOutputStream {
+
+  private static String testKeyString = "test";
+
+  @Test
+  public void testRATISKeyOutputStreamExpiryTime() throws Exception {
+    KeyOutputStream keyOutputStream =
+        createRATISKeyOutputStream();
+    byte[] data = testKeyString.getBytes(UTF_8);
+    keyOutputStream.write(data);
+
+    Assert.assertEquals(3,
+        keyOutputStream.getExcludeList().getDatanodes().size());
+
+    ExcludeList excludeList = spy(keyOutputStream.getExcludeList());
+    when(excludeList.getExpiryTime()).thenReturn(300 * 1000L);
+    doReturn(true).when(excludeList)
+        .isExpired(anyLong()); // mock DN in exclude list expires
+    keyOutputStream.getBlockOutputStreamEntryPool().setExcludeList(excludeList);
+    Assert.assertEquals(0,
+        keyOutputStream.getExcludeList().getDatanodes().size());
+  }
+
+  private KeyOutputStream createRATISKeyOutputStream() throws Exception {
+    OpenKeySession openKeySession = mock(OpenKeySession.class);
+    OmKeyInfo omKeyInfo =  new OmKeyInfo.Builder()
+        .setVolumeName("testvolume")
+        .setBucketName("testbucket")
+        .setKeyName("testKey")
+        .build();
+    when(openKeySession.getKeyInfo()).thenReturn(omKeyInfo);
+
+    XceiverClientFactory xceiverClientManager
+        = mock(XceiverClientFactory.class);
+
+    OzoneManagerClientProtocol ozoneManagerClientProtocol
+        = mock(OzoneManagerClientProtocol.class);
+
+    OzoneClientConfig clientConfig = spy(new OzoneClientConfig());
```

Comment:
we do not need spy ozone config, just need set config value with new configuration

Reply:
sure! removed spy.